src/compiler/scanner.rs

```diff
@@ -7,11 +7,16 @@ use std::iter::FromIterator;
 
 use crate::{CharacterCodes, SyntaxKind, TokenFlags};
 
+pub fn token_is_identifier_or_keyword(token: SyntaxKind) -> bool {
+    token >= SyntaxKind::Identifier
+}
+
 lazy_static! {
     static ref text_to_keyword_obj: HashMap<String, SyntaxKind> =
         HashMap::from_iter(IntoIter::new([
             ("const".to_string(), SyntaxKind::ConstKeyword),
             ("false".to_string(), SyntaxKind::FalseKeyword),
+            ("interface".to_string(), SyntaxKind::InterfaceKeyword),
```

Comment:
It doesn't really matter, but I bet this could be a `HashMap<'static &str, SyntaxKind>` and you could avoid the conversions.

Reply:
I want to say I tried that and ran into something but can't really remember